hyperspy/tests/io/test_hdf5.py

```diff
@@ -658,8 +658,9 @@ def setup_method(self):
     
     @pytest.mark.parametrize("compression", (None, "gzip", "szip", "lzf", "blosc", ))
     def test_compression(self, compression, tmp_path):
-        if compression == "szip" and sys.platform == "win32":
-            pytest.skip('szip not supported on windows')
+        if compression == "szip": # Support for szip is dependent on HDF5 install
```

Comment:
Why is it necessary to test the `szip` compression filters? This is testing the hdf5 install, not hyperspy... and we don't need to test this! I would simply remove using this compression filters from the list of parameters and remove the corresponding function in `hspy.py`.